test/Extensions/ServiceBus.Tests/Streaming/EHCustomBatchContainerTests.cs

```diff
@@ -0,0 +1,111 @@
+                hostBuilder
+                    .AddMemoryGrainStorage("PubSubStore")
+                    .AddEventHubStreams(StreamProviderName, b =>
+                    {
+                        b.ConfigureCacheEviction(ob => ob.Configure(options =>
+                        {
+                            options.DataMaxAgeInCache = TimeSpan.FromSeconds(5);
+                            options.DataMinTimeInCache = TimeSpan.FromSeconds(0);
+                            options.MetadataMinTimeInCache = TimeSpan.FromMinutes(1);
+                        }));
+                        b.ConfigureEventHub(ob => ob.Configure(options =>
+                        {
+                            options.ConnectionString = TestDefaultConfiguration.EventHubConnectionString;
+                            options.ConsumerGroup = EHConsumerGroup;
+                            options.Path = EHPath;
+                        }));
+                        b.UseAzureTableCheckpointer(ob => ob.Configure(options =>
+                        {
+                            options.ConnectionString = TestDefaultConfiguration.DataConnectionString;
+                            options.PersistInterval = TimeSpan.FromSeconds(10);
+                        }));
+                        b.UseDataAdapter((sp, n) => ActivatorUtilities.CreateInstance<CustomDataAdapter>(sp));
+                    });
+            }
+        }
+
+        private class MyClientBuilderConfigurator : IClientBuilderConfigurator
+        {
+            public void Configure(IConfiguration configuration, IClientBuilder clientBuilder)
+            {
+                clientBuilder
+                    .AddEventHubStreams(StreamProviderName, b =>
+                    {
+                        b.ConfigureEventHub(ob => ob.Configure(options =>
+                        {
+                            options.ConnectionString = TestDefaultConfiguration.EventHubConnectionString;
+                            options.ConsumerGroup = EHConsumerGroup;
+                            options.Path = EHPath;
+                        }));
+                        b.UseDataAdapter((sp, n) => ActivatorUtilities.CreateInstance<CustomDataAdapter>(sp));
+                    });
+            }
+        }
+
+        private class CustomDataAdapter : EventHubDataAdapter
+        {
+            public CustomDataAdapter(SerializationManager serializationManager)
+                : base(serializationManager)
+            {
+            }
+
+            public override IBatchContainer GetBatchContainer(ref CachedMessage cachedMessage)
+                => new CustomBatchContainer(base.GetBatchContainer(ref cachedMessage));
+
+            protected override IBatchContainer GetBatchContainer(EventHubMessage eventHubMessage)
+                => new CustomBatchContainer(base.GetBatchContainer(eventHubMessage));
+
+            //public override string GetPartitionKey(Guid streamGuid, string streamNamespace) => Guid.Empty.ToString();
```

Comment:
The commented code here might make people reading this wonder if it was/is important
```suggestion
```